### PR TITLE
Card style updated, created a single shared component [Fixes #2251]

### LINKED ISSUE #2251: Translation progress card style update
**Is your feature request related to a problem? Please describe.**

These card styles are outdated:
https://ethereum.org/en/contributing/translation-program/#in-progress

**Describe the solution you'd like**

Similar to #2076 & #2195 - let's update these card styles to match the one on the Languages page:
https://ethereum.org/en/languages/

Given that these card styles are now in 3 separate components, let's see if we can consolidate these into a single shared component in SharedStyledComponents.js 

**Describe alternatives you've considered**

**Additional context**

We should wait until #2227 is merged before moving on this.

### PR DESCRIPTION
**Pull request by Anuragtech02**
## Description
- [x] Updated card style on https://ethereum.org/en/contributing/translation-program/ to match the one on the Languages page: https://ethereum.org/en/languages/
- [x]  Created a single shared component in SharedStyledComponents.js

Created a single shared component CardItem in SharedStyledComponents and replaced the cards on `/languages` and `/about` ( #2076 & #2195 ) with the new shared component. Doing this also unified card hover transition.

## Related Issue #2251
These card styles were outdated:
https://ethereum.org/en/contributing/translation-program/#in-progress so @samajammin I updated them to similar ones like on `/languages` and `/about`.

### CODE REVIEW

File: src/components/SharedStyledComponents.js [hunk 1]
```diff
@@ -443,3 +445,28 @@ export const dropdownIconContainerVariant = {
...
+export const CardItem = styled(GatsbyLink)`
+  text-decoration: none;
+  margin: 1rem 1rem 1rem 0;
+  padding: 1rem;
+  flex: 1 1 200px;
+  list-style: none;
+  border-radius: 2px;
+  width: 100%;
+  border: 1px solid ${(props) => props.theme.colors.lightBorder};
+  ${"" /* box-shadow: ${(props) => props.theme.colors.tableBoxShadow}; */}
```
Comment (samajammin) [inline src/components/SharedStyledComponents.js:460]: Let's just delete this line given that it's commented out.

File: src/components/SharedStyledComponents.js [hunk 2]
```diff
@@ -443,3 +445,28 @@ export const dropdownIconContainerVariant = {
   },
   closed: { rotate: -90, y: 0 },
 }
+
+// Common styled item card for languages/about
+
+export const CardItem = styled(GatsbyLink)`
```
Comment (samajammin) [inline src/components/SharedStyledComponents.js:451]: I *believe* we can just use `Link` here vs. `GatsbyLink`. In `Link` component we check to see if language path has been explicitly set: https://github.com/ethereum/ethereum-org-website/blob/dev/src/components/Link.js#L131

File: src/components/SharedStyledComponents.js [hunk 3]
```diff
@@ -3,6 +3,8 @@ import styled from "styled-components"
 import { Mixins } from "../theme"
 import Card from "./Card"
 import Link from "./Link"
+//Imported for LangItem
+import { Link as GatsbyLink } from "gatsby"
```
Comment (samajammin) [inline src/components/SharedStyledComponents.js:7]: I don't think this is needed. See comment below.

File: src/components/Roadmap.js [hunk 4]
```diff
@@ -9,6 +9,9 @@ import { FakeLinkExternal, H2, H3 } from "./SharedStyledComponents"
 
 import { translateMessageId } from "../utils/translations"
 
+//Import for Item / Card
```
Comment (samajammin) [inline src/components/Roadmap.js:12]: No need for comment, please remove 😄

File: src/components/SharedStyledComponents.js [hunk 5]
```diff
@@ -443,3 +445,28 @@ export const dropdownIconContainerVariant = {
...
   closed: { rotate: -90, y: 0 },
 }
+
+// Common styled item card for languages/about
+
+export const CardItem = styled(GatsbyLink)`
+  text-decoration: none;
+  margin: 1rem 1rem 1rem 0;
+  padding: 1rem;
+  flex: 1 1 200px;
```
Comment (samajammin) [inline src/components/SharedStyledComponents.js:455]: I don't think this quite works.

It looks great for the in-progress translation cards:
https://share.getcloudapp.com/yAu6QOkE

But it warps the styles of the language cards (see your preview deploy: https://deploy-preview-2481--ethereumorg.netlify.app/en/languages/): 
https://share.getcloudapp.com/bLug5EBE
Those cards likely need that default `width: 240px;` property. Perhaps you overwrite these base styles in that file (e.g. `LangItem = styled(CardItem)`...

**Review by samajammin — CHANGES_REQUESTED** [review]
Thanks for the PR! A couple comments. Please let me know if you have any questions 😄

File: src/components/SharedStyledComponents.js (hunk 5 above)
Comment (Anuragtech02) [reply src/components/SharedStyledComponents.js:455]: Got it! I'll overwrite it for the language card. Also do let me know if you want to get it changed for the work in progress cards on `/about`.

**Comment by Anuragtech02** [comment]
Thanks @samajammin !, I've made the changes

File: src/components/TranslationsInProgress.js [hunk 6]
```diff
@@ -1,33 +1,17 @@
...
   margin-bottom: 2rem;
   display: flex;
   flex-wrap: wrap;
 `
-
-const LangItem = styled(GatsbyLink)`
-  text-decoration: none;
-  margin: 1rem 1rem 1rem 0;
-  padding: 1rem;
+const LangItem = styled(CardItem)`
```
Comment (wackerow) [inline src/components/TranslationsInProgress.js:13]: This renders with an extra "external" arrow glyph. By switching this to use `CardItem`, it now treats this link differently and adds an arrow glyph. This is already accommodated for in the `FakeExternalLink` which puts the arrow next to the appropriate text.

tl;dr - Add `hideArrow` attribute to L42 to hide extra arrow:
`<LangItem to={url} key={lang.code} hideArrow>`

File: src/components/SharedStyledComponents.js [hunk 7]
```diff
@@ -443,3 +443,27 @@ export const dropdownIconContainerVariant = {
...
   closed: { rotate: -90, y: 0 },
 }
+
+// Common styled item card for languages/about
+
+export const CardItem = styled(Link)`
+  text-decoration: none;
+  margin: 1rem 1rem 1rem 0;
+  padding: 1rem;
+  flex: 1 1 240px;
```
Comment (wackerow) [inline src/components/SharedStyledComponents.js:453]: This causes the bottom row of the Translations page to have ~1-3 cards that get stretched to fill the whole row (depending on screen width).
`flex: 0 1 240px;` should prevent this

**Review by wackerow — CHANGES_REQUESTED** [review]
@Anuragtech02 Thanks again for the PR =) Spotted two things... one is the redundant external arrow, the other being the bottom row expanding full width (see inline comments). Otherwise should be good to go!

**Comment by Anuragtech02** [comment]
Thanks @wackerow ! I've added the hideArrow attr and set the flex grow to zero. I've also removed a redundant media query which was trying to set card width to 100% (which was already 100%).

**Review by samajammin — CHANGES_REQUESTED** [review]
Almost there @Anuragtech02!

Looking at your preview deploy (https://deploy-preview-2481--ethereumorg.netlify.app/en/languages/) in mobile:
Image 2021-03-03 at 2 29 39 PM

The cards are no longer full width, like they are in production:
Image 2021-03-03 at 2 30 15 PM

I suspect the media query or the flex grow you deleted is actually needed on the language page items.

**Comment by Anuragtech02** [comment]
> Almost there @Anuragtech02!
> 
> Looking at your preview deploy (https://deploy-preview-2481--ethereumorg.netlify.app/en/languages/) in mobile:
> Image 2021-03-03 at 2 29 39 PM
> 
> The cards are no longer full width, like they are in production:
> Image 2021-03-03 at 2 30 15 PM
> 
> I suspect the media query or the flex grow you deleted is actually needed on the language page items.

@samajammin  The media query was not the issue, it was the flex-grow. I added a media query for mobile screens which sets flex grow to 1 which now solves the issue for us.

**Review by wackerow — APPROVED** [review]
👍 @Anuragtech02 Looks good! Added one last little tweak to remove to right margin on mobile screens to even it out. Thanks again!
cc @samajammin

**Review by samajammin — APPROVED** [review]
Thanks for following through on this ❤️

**Comment by samajammin** [comment]
@all-contributors please add @Anuragtech02 for code.
